crates/ruff_db/src/vendored.rs

```diff
@@ -0,0 +1,351 @@
+use std::cell::RefCell;
+use std::io::{self, Read};
+use std::sync::{Mutex, MutexGuard};
+
+use itertools::Itertools;
+use zip::{read::ZipFile, ZipArchive};
+
+use crate::metadata::FileRevision;
+pub use path::{VendoredPath, VendoredPathBuf};
+
+pub mod path;
+
+type Result<T> = io::Result<T>;
+
+#[derive(Debug)]
+pub struct VendoredFileSystem {
```

Comment:
It might be helpful to add some documentation to this struct, even if it's very brief